File: dataset/human/java/file_245.java
import java.util.*;
public class Main {
	public static void main(String args[]) {
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		int A[] = new int[N];
		int cnt = 0;
		int min = Integer.MAX_VALUE;
		long sum = 0;
		for(int i = 0; i < N; i++) {
			A[i] = sc.nextInt();
			if(A[i] < 0) {
				cnt++;
			}
			min = Math.min(Math.abs(min), Math.abs(A[i]));
			sum += Math.abs(A[i]);
		}
		
		if(cnt % 2 == 0) {
			System.out.println(sum);
		} else {
			System.out.println(sum - Math.abs(min) * 2);
		}
	}
}

File: dataset/human/c/file_245.c
#include <stdio.h>

int n;
int a[100000];
unsigned long int sum = 0;

int abs(int a) { if(a > 0) return a; return (-1)*a; }

int main(void)
{
	int i, min = 1000000000, cnt = 0;
	scanf("%d%*c", &n);
	for(i=0;i<n;i++) {
		scanf("%d%*c", &a[i]);
		if(a[i] <= 0) cnt++;
		int tmp = abs(a[i]);
		if(min > tmp) min = tmp;
		sum += tmp;
	}

	if(cnt%2 == 0) printf("%ld", sum);
	else printf("%ld", sum - min*2);
	
	return 0;
}
